ICoder.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace objected_oriented_programming
+{
+    interface ICoder
+    {
+        public void Encode();
+        public void Decode();
+    }
```

Comment:
Модификаторы доступа не там поставили. Внутри интерфейса для его членов модификаторы доступа не ставятся - все члены и так считаются `public`.
```suggestion
    internal interface ICoder
    {
        void Encode();
        void Decode();
    }
```